prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -34,59 +34,86 @@ object FhirPathUtils : Logging {
     }
 
     /**
-     * Gets a FHIR base resource from the given [expressionNode] using [bundle] and starting from a specific [focusResource].
+     * Gets a FHIR base resource from the given [expression] using [bundle] and starting from a specific [focusResource].
      * [focusResource] can be the same as [bundle] when starting from the root.
      * [appContext] provides custom context (e.g. variables) used for the evaluation.
      */
     fun evaluate(
         appContext: CustomContext?,
         focusResource: Base,
         bundle: Bundle,
-        expressionNode: ExpressionNode
+        expression: String
     ): List<Base> {
-        val resources = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
-        return resources.map {
-            if (it.hasType("Reference") && (it as Reference).resource != null) {
-                it.resource as Base
-            } else it
+        val retVal = try {
+            val resources = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, parsePath(expression))
```

Comment:
Why are we calling this `defaultPathEngine`?  There is no way to provide an alternate path engine - is that for future phases?

Reply:
It can be called just path engine.  I just grabbed the name Gage has, but I think you are right and there will not be any other type of engine